packages/ses/src/reporting.js

```diff
@@ -0,0 +1,104 @@
+import { globalThis, TypeError } from './commons.js';
+import { assert } from './error/assert.js';
+
+/**
+ * @import {Reporter, GroupReporter} from './reporting-types.js'
+ */
+
+/**
+ * Creates a suitable reporter for internal errors and warnings out of the
+ * Node.js console.error to ensure all messages to go stderr, including the
+ * group label.
+ * Accounts for the extra space introduced by console.error as a delimiter
+ * between the indent and subsequent arguments.
+ *
+ * @param {(...message: Array<any>) => void} print
+ */
+const makeReportPrinter = print => {
+  let indent = false;
+  /** @param {Array<any>} args */
+  const printIndent = (...args) => {
+    if (indent) {
+      print(' ', ...args);
+    } else {
+      print(...args);
+    }
+  };
+  return /** @type {GroupReporter} */ ({
+    warn(...args) {
+      printIndent(...args);
+    },
+    error(...args) {
+      printIndent(...args);
+    },
+    groupCollapsed(...args) {
+      assert(!indent);
+      print(...args);
+      indent = true;
+    },
+    groupEnd() {
+      indent = false;
+    },
+  });
+};
+
+const mute = () => {};
+
+/**
+ * @param {"platform" | "console" | "none"} reporting
+ */
+export const chooseReporter = reporting => {
+  if (reporting === 'none') {
+    return makeReportPrinter(mute);
+  }
+  if (reporting !== 'platform' && reporting !== 'console') {
+    throw new TypeError(`Invalid lockdown reporting option: ${reporting}`);
+  }
+  if (
+    reporting === 'console' ||
+    globalThis.window === globalThis ||
+    globalThis.importScripts !== undefined
+  ) {
+    return console;
+  }
+  if (globalThis.console !== undefined) {
+    // On Node.js, we send all feedback to stderr, regardless of purported level.
+    return makeReportPrinter(globalThis.console.error);
```

Comment:
I'm just a bit surprised that `console.error` is thisless, i.e., that it doesn't need to be bound to `console` first. I tried it, and that is indeed the case in Node. Curious though if it is specified that way, by those who purport to standardize `console`?

Reply:
Back in Safari 2, `console.error.call(null)` caused a segmentation fault for the whole browser.

Reply:
WOW! Please report. From previous history, I know they'll be interested.

Reply:
Oh, Safari 2! Nevermind. Resolving.